src/lua/httpc.lua

```diff
@@ -32,8 +32,18 @@
 local driver = package.loaded.http.client
```

Comment:
Regarding the commit message (1st patch):

> strings, numbers and booleans will be converted to a string using "tostring()" function, tables, userdata and cdata will be serialized by calling a metamethod
`__serialize`

I cited this clause for a context, but the comment is about the documentation request at whole.

It was unclear for me from the documentation request that only cdata, userdata, table values (including tarantool's own types like decimal) are goes through standard or user-defined encoders. Everything else has the one encoding way: string, number, boolean goes to `tostring()`, nil becomes an empty string.

I can easily imagine something like:

```lua
local http_client = require('http.client')
http_client:post('https://<...>', 42, {
    headers = {
        ['Content-Type'] = 'application/msgpack',
    }
})
```

With expectation that it'll be encoded as msgpack.

It seems, your wording assumes it by crossing from '`__serialize` is used' to 'the serialization formats are <...>'. I would suggest to emphasise the difference. For example:

> <...> strings, numbers and booleans will be converted to a string using "tostring()" function, nil means an empty body.
>
> Other Lua types are special in the sense that they have no univocal string representation. A body value of table, userdata or cdata type will be passed through an encoder function. The function is choosen based on <...>. If no content type is provided, <...>.

(I hope it'll allow to avoid the misunderstanding in a future.)

Reply:
Sure, thank you!